core/indexer/elasticsearch.go

```diff
@@ -86,43 +96,57 @@ func (ei *elasticIndexer) SaveBlock(
 		return
 	}
 
-	go ei.database.SaveHeader(headerHandler, signersIndexes)
+	go ei.database.SaveHeader(headerHandler, signersIndexes, body, notarizedHeadersHashes)
 
 	if len(body.MiniBlocks) == 0 {
 		log.Debug("indexer", "error", ErrNoMiniblocks.Error())
 		return
 	}
 
-	if ei.options.TxIndexingEnabled {
-		go ei.database.SaveTransactions(body, headerHandler, txPool, ei.shardCoordinator.SelfId())
-	}
-}
+	go ei.database.SaveMiniblocks(headerHandler, body)
 
-// SaveMetaBlock will index a meta block in elastic search
-func (ei *elasticIndexer) SaveMetaBlock(header data.HeaderHandler, signersIndexes []uint64) {
-	if check.IfNil(header) {
-		log.Debug("indexer: nil header", "error", ErrNoHeader.Error())
-		return
+	if ei.options.TxIndexingEnabled {
+		go ei.database.SaveTransactions(body, headerHandler, txPool, headerHandler.GetShardID())
 	}
-
-	go ei.database.SaveHeader(header, signersIndexes)
 }
 
 // SaveRoundInfo will save data about a round on elastic search
 func (ei *elasticIndexer) SaveRoundInfo(roundInfo RoundInfo) {
 	ei.database.SaveRoundInfo(roundInfo)
 }
 
-//SaveValidatorsPubKeys will send all validators public keys to elastic search
-func (ei *elasticIndexer) SaveValidatorsPubKeys(validatorsPubKeys map[uint32][][]byte) {
+func (ei *elasticIndexer) epochStartEventHandler() epochStart.EpochStartHandler {
+	subscribeHandler := notifier.NewHandlerForEpochStart(func(hdr data.HeaderHandler) {
+		currentEpoch := hdr.GetEpoch()
+		validatorsPubKeys, err := ei.coordinator.GetAllEligibleValidatorsPublicKeys(currentEpoch)
+		if err != nil {
+			log.Warn("GetAllEligibleValidatorPublicKeys for epoch 0 failed", "error", err)
```

Comment:
is not epoch 0 anymore, please change log message

Reply:
done.